packages/plugin-dev/src/browser/hosted-plugin-manager-client.ts

```diff
@@ -265,7 +265,7 @@ export class HostedPluginManagerClient {
         if (UriSelection.is(result)) {
             if (await this.hostedPluginServer.isPluginValid(result.uri.toString())) {
                 this.pluginLocation = result.uri;
-                this.messageService.info('Plugin folder is set to: ' + result.uri.toString());
+                this.messageService.info('Plugin folder is set to: ' + decodeURI(result.uri.toString()));
```

Comment:
You can use ` result.uri.toString(true)` instead.

Reply:
thinking again, `LabelProvider.getLongName` should be used to a display string from a URI to align with the rest of UI